python/nav/models/manage.py

```diff
@@ -833,6 +833,30 @@ class Meta(object):
     def __str__(self):
         return self.serial or ''
 
+    def get_related_objects(self):
+        modules = self.module_set.all()
+        power_supplies_or_fans = self.powersupplyorfan_set.all()
+        netbox_entities = self.netboxentity_set.all()
+        return modules or power_supplies_or_fans or netbox_entities
+
+    def get_preferred_related_object(self):
+        related_objects = self.get_related_objects()
+        if not related_objects:
+            return None
+        if len(related_objects) > 1:
+            _logger.info(
+                "Device.get_related_objects(): %s weirdly appears to have "
+                "duplicate entities, returning just one",
```

Comment:
Please use the term "related objects" rather than "entities", since the latter can easily be confused with entities from `NetboxEntity`, which is a lot more specific...